kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -196,7 +196,7 @@ return {
           -- please see https://github.com/Kong/kong/issues/5763
           red:eval([[
             local key, value, expiration = KEYS[1], tonumber(ARGV[1]), ARGV[2]
-            if redis.call("incrby", key, value) == value then
+            if redis.call("incrby", key, value) == 1 then
```

Comment:
Still I think this is implying that if key does not exists OR key value is 0, then we expire the key.
But do we want to expire a key that's value =0?
The test is comparing db size so that's probably why it's failing.

Reply:
Thanks for reminding me.
The old logic is that the expiration is set only when the key does not exist. But even if I do that, it doesn't solve this test.